src/cmd/palette/command.rs

```diff
@@ -26,13 +26,26 @@ pub struct PaletteImportArgs {
 
 /// Palette import command
 pub fn palette_import(g_args: &GlobalArgs, args: &PaletteImportArgs) -> Result<(), Error> {
-	let png_bytes = common::read_file_bin(&args.input_file)?;
-	let pal_config = VeraPaletteLoadConfig {
-		direct_load: true,
-		include_defaults: false,
-		sort: false,
+	let is_gpl = args.input_file.ends_with("gpl") || args.input_file.ends_with("GPL");
+	let palette = match is_gpl {
+		true => {
+			let pal_config = VeraPaletteLoadConfig {
+				direct_load: true,
+				include_defaults: false,
+				sort: false,
+			};
+			VeraPalette::derive_from_gpl(&args.id, &args.input_file, &pal_config).expect("Error")
```

Comment:
Just return the error here as originally. The `?` operator at the end of something that returns a Result either unwraps the result or returns from the function with an error if it fails, which will work so long as the error type is understood by the enclosing function. Unfortunately rust's error handling is a bit all over the place, and you need to ensure error handling is consistent throughout or you start getting unhelpful panics and difficulty tracking down errors.

Reply:
Fixed.